src/metrics.py

```diff
@@ -0,0 +1,32 @@
+#  Copyright 2023 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Models and functions for the metric events."""
+
+from pydantic import BaseModel, conint
+
+
+class Event(BaseModel):
+    """Base class for metric events."""
+
+    timestamp: conint(ge=0)
+    """The UNIX timestamp of the event."""
+
+
+class RunnerInstalled(Event):
+    """Metric event for when a runner is installed."""
+
+    name: str = "runner_installed"
+    """The name of the event."""
+    flavor: str
+    """The flavor of the runner."""
+    duration: conint(ge=0)
+    """The duration of the installation in seconds."""
```

Comment:
What are the docstrings below the attributes? Are the intended to be attribute descriptions? Usually those are done like this in the class docstring:

```python
class RunnerInstalled(Event):
    """
    Metric event for when a runner is installed.

    Attrs:
        name: The name of the event.
        ....
```